### ISSUE TITLE
Upcoming changes to GYP

### ISSUE THREAD

**Issue by hashseed** [nodejs/node-v8#38]
As previously described in [this document](https://docs.google.com/document/d/1gvHuesiuvLzD6X6ONddxXRxhODlOJlxgfoTNZTlKLGA/edit) and tracked in [these](https://github.com/nodejs/CTC/issues/2) Node.js [issues](https://github.com/nodejs/TSC/issues/411) and this [V8 issue](https://bugs.chromium.org/p/v8/issues/detail?id=6105), V8 has implemented a GYP/GN bridge to allowing building V8 with GN but keep the rest of Node core on GYP.

The Node.js part is implemented [here](https://github.com/v8/node/commit/6d9eb3f52f185532e9270a0917dda814e8b8d41d) and updated [here](https://github.com/v8/node/commits/vee-eight-lkgr). That adds a new configure flag `--build-v8-with-gn`, which is tested on Linux and Mac. Running configure with that flag fetches the necessary dependencies and subsequent make builds V8 with GN.

I'm currently moving GYP files into `v8/gypfiles`. Then I'll simply delete the folder to conclude the handover. This will immediately affect the canary branch. You could either switch to `--build-v8-with-gn`, or you need to take over ownership over V8's GYP files.

**Comment by mhdawson** [comment]
Sounds like we need a decision on:

>You could either switch to --build-v8-with-gn, or you need to take over ownership over V8's GYP files.

labeling so that it's on the agenda a week from now unless we resolve through GitHub or the discussion with the V8 team after this weeks TSC meeting.

**Comment by mhdawson** [comment]
@hashseed  I see it's tested on Linux from the doc, can we test it across platforms with one of the existing CI jobs?

**Comment by hashseed** [comment]
@mhdawson I'm not sure I expressed myself understandably. So let me try again.

`--build-v8-with-gn` is a crutch that we at V8 use for continuously test V8 against Node.js without having to maintain gyp files. It is tested for Linux and Mac, but not yet on Windows.

The canary build at node-v8 is similar, and unless you want to update gyp files on every canary, you could use the same crutch. For updating V8 on master I would recommend to keep building V8 with gyp that Node.js would inherit as soon as we delete them in V8 upstream. That entails maintaining gyp files on every update.

I can explain in detail during the V8/TSC call.

**Comment by mhdawson** [comment]
@hashseed thanks for the clarification, got it.

**Comment by rvagg** [comment]
We need to have gn binaries on all build machines for this to work right? And for people building from source we have to explain how to get gn installed as well. Is it in an easier to fetch form these days or still bundled up with other stuff?

**Comment by hashseed** [comment]
We have all of this implemented on our [fork](https://github.com/v8/node/commits/vee-eight-lkgr). There I added a new configure flag so that you can run `./configure --build-v8-with-gn`. This will fetch the necessary dependencies. All you then need to do is `make`. I'll upstream this once Node.js upgrades to V8 6.6.

**Comment by hashseed** [comment]
gyp configs [have been removed](https://chromium.googlesource.com/v8/v8/+/2c8663aa1a2d7e411bb2759dfe31d289990db135) upstream.

**Comment by targos** [comment]
Thanks for the heads up!
I updated the canary update script to take this into account.

**Comment by targos** [comment]
I think this can be closed
